forc-plugins/forc-doc/src/doc.rs

```diff
@@ -119,11 +118,76 @@ impl Document {
         Ok(())
     }
 }
-impl crate::render::Renderable for Document {
-    fn render(self, declaration_engine: Arc<DeclarationEngine>) -> Box<dyn RenderBox> {
+impl Renderable for Document {
+    fn render(self) -> Box<dyn RenderBox> {
         box_html! {
-            : self.item_header.render(declaration_engine.clone());
-            : self.item_body.render(declaration_engine);
+            : self.item_header.render();
+            : self.item_body.render();
+        }
+    }
+}
+pub(crate) type ModulePrefix = String;
+#[derive(Clone)]
+pub(crate) struct ModuleInfo(pub(crate) Vec<ModulePrefix>);
+impl ModuleInfo {
+    /// The current module.
+    pub(crate) fn location(&self) -> &str {
+        self.0
+            .last()
+            .expect("There will always be at least the project name")
+    }
+    /// The location of the parent of the current module.
+    ///
+    /// To be used in path navigation between modules.
+    pub(crate) fn _parent(&mut self) -> &str {
+        self.0.pop();
+        self.location()
+    }
```

Comment:
I guess we are not using this function, maybe we can remove it?

```suggestion
```

Reply:
So currently, all paths lead back to the `AllDoc` (`all.html`) path. But this isn't the end goal. How it _should_ be is that if you are on an item page the `parent` module should be the desired path. I'd like to keep this fn since my follow up PR will be tackling the aforementioned problem if that's ok!

Reply:
sounds reasonable enough to me.

Reply:
For future reference, we probably don't want this method to mutate `self` as it currently does, otherwise consecutive calls to `_parent` will produce different results. We might want to return an `Option` too, as if this is the root module, I guess there won't be a parent?